libs/angular-generator/src/lib/theming/theming-models.ts

```diff
@@ -0,0 +1,45 @@
+import { ExportSpecifierType, GeneratedFile } from "@ui5/webcomponents-wrapper";
+import { NodeFsImplementation } from "@ui5/webcomponents-wrapper-fs-commit";
+import { format } from "prettier";
+
+export class Ui5ThemingModels extends GeneratedFile {
+  constructor() {
+    super();
+    this.move('ui5-theming.models.ts');
+    this.addExport({
+      local: '*',
+      exported: '*',
+      types: [ExportSpecifierType.Variable],
+    });
+  }
+
+  getThemeNames(): string[] {
+    const nodeFs = new NodeFsImplementation();
+
+    const fdPath = nodeFs.normalize('node_modules/fundamental-styles/dist/theming/');
+
+    return nodeFs.queryFiles(`${fdPath}**.css`, []).map((themePath) => nodeFs.basename(themePath, nodeFs.extname(themePath)));
+  }
+
+  override getCode(): string {
+    this.addImport(['InjectionToken'], '@angular/core');
```

Comment:
I moved this addImports into other functions, after constructor step, getCode might be called several times and it is just redundant operations

Reply:
Can't imagine a case when getCode would be called multiple times for one class instance. Could you please elaborate?